### PR TITLE
Timeout connections to observers

### PR DESCRIPTION
**Pull request by obycode**
Without a timout for these connections, the node can get stuck here and it is not obvious what is happening. With this timeout, we will see log messages that will point to the problem.

### CODE REVIEW

File: testnet/stacks-node/src/event_dispatcher.rs [hunk 1]
```diff
@@ -326,19 +327,26 @@ impl EventObserver {
             req.set_body(body);
 
             let response = async_std::task::block_on(async {
-                let stream = match TcpStream::connect(self.endpoint.clone()).await {
-                    Ok(stream) => stream,
-                    Err(err) => {
+                match async_std::future::timeout(
+                    connection_timeout,
```
Comment (jcnelson) [inline testnet/stacks-node/src/event_dispatcher.rs:331]: If we're going to change this code path, can we instead please just drop async/await?  It makes using gdb harder than it needs to be, and it pulls in a crap-ton of dependencies

Comment (kantai) [reply testnet/stacks-node/src/event_dispatcher.rs:331]: No, I'd prefer that this continue to use async/await as it is now.

**Comment by obycode** [comment]
> I'm a little wary of this change -- it could lead to duplicate events being received by an event listener and 5 seconds seems like a kind of arbitrary (and short-ish) timeout. Some of the events can get kind of beefy, and while 5 seconds is a good chunk of time, I'm not convinced that this wouldn't lead to a good chunk of false timeouts.

Sure, that's a valid concern. What if I just timeout the TCP stream connect? I think that's where it should be stuck if the observer has crashed or something.

**Comment by obycode** [comment]
I made that change in 19076c1e5.

**Review by kantai — APPROVED** [review]

File: testnet/stacks-node/src/event_dispatcher.rs (hunk 1 above)
Comment (obycode) [reply testnet/stacks-node/src/event_dispatcher.rs]: If we decide to do that, it should be in a separate refactoring PR, but I'd prefer not to do that before Nakamoto.

**Review by jferrant — APPROVED** [review]

**Pull request merged**